backend/services/nudgeScheduler.js

```diff
@@ -0,0 +1,77 @@
+const { supabase } = require("./supabaseclient");
+const { getUsersActiveWindow } = require("./activityAnalysis");
+const { sendNudgeToUser } = require("../websocket/websocketServer");
+
+async function findNotificationsNeedingNudge() {
+  try {
+    //subtracing 3 hours from current time
+    const threeHoursAgo = new Date(
+      Date.now() - 3 * 60 * 60 * 1000
```

Comment:
Can we move this to a constant? That will make it easier edit it if needed. And considering that you're using many time constants, maybe just create a separate file with some general ones like MS_IN_MINUTE, MS_IN_HOUR, etc